src/openrct2-ui/windows/ScenarioSelect.cpp

```diff
@@ -82,76 +81,59 @@ enum {
+    MakeWidget({ TabWidth + 3, WidgetsStart + 1 }, { WW - SidebarWidth, 276 }, WindowWidgetType::Scroll, WindowColour::Secondary, SCROLL_VERTICAL), // level list
     WIDGETS_END,
 };
+// clang-format on
+static void WindowScenarioselectInitTabs(WindowBase* w);
+static void DrawCategoryHeading(WindowBase* w, DrawPixelInfo* dpi, int32_t left, int32_t right, int32_t y, StringId stringId);
+static void InitialiseListItems(WindowBase* w);
+static bool IsScenarioVisible(WindowBase* w, const ScenarioIndexEntry* scenario);
+static bool IsLockingEnabled(WindowBase* w);
 
-static constexpr const StringId ScenarioOriginStringIds[] = {
-    STR_SCENARIO_CATEGORY_RCT1,
-    STR_SCENARIO_CATEGORY_RCT1_AA,
-    STR_SCENARIO_CATEGORY_RCT1_LL,
-    STR_SCENARIO_CATEGORY_RCT2,
-    STR_SCENARIO_CATEGORY_RCT2_WW,
-    STR_SCENARIO_CATEGORY_RCT2_TT,
-    STR_SCENARIO_CATEGORY_UCES,
-    STR_SCENARIO_CATEGORY_REAL_PARKS,
-    STR_SCENARIO_CATEGORY_EXTRAS_PARKS,
-    STR_SCENARIO_CATEGORY_OTHER_PARKS,
-};
-
-static void WindowScenarioselectInitTabs(WindowBase *w);
-
-static void WindowScenarioselectClose(WindowBase *w);
-static void WindowScenarioselectMouseup(WindowBase *w, WidgetIndex widgetIndex);
-static void WindowScenarioselectMousedown(WindowBase *w, WidgetIndex widgetIndex, Widget* widget);
-static void WindowScenarioselectScrollgetsize(WindowBase *w, int32_t scrollIndex, int32_t *width, int32_t *height);
-static void WindowScenarioselectScrollmousedown(WindowBase *w, int32_t scrollIndex, const ScreenCoordsXY& screenCoords);
-static void WindowScenarioselectScrollmouseover(WindowBase *w, int32_t scrollIndex, const ScreenCoordsXY& screenCoords);
-static void WindowScenarioselectInvalidate(WindowBase *w);
-static void WindowScenarioselectPaint(WindowBase *w, DrawPixelInfo *dpi);
-static void WindowScenarioselectScrollpaint(WindowBase *w, DrawPixelInfo *dpi, int32_t scrollIndex);
+static bool _titleEditor = false;
+static bool _disableLocking{};
+static std::function<void(std::string_view)> _callback;
+static std::vector<ScenarioListItem> _listItems;
 
 static bool ScenarioSelectUseSmallFont()
 {
     return ThemeGetFlags() & UITHEME_FLAG_USE_ALTERNATIVE_SCENARIO_SELECT_FONT;
 }
 
-static WindowEventList window_scenarioselect_events([](auto& events)
+static int32_t GetScenarioListItemSize()
 {
-    events.close = &WindowScenarioselectClose;
-    events.mouse_up = &WindowScenarioselectMouseup;
-    events.mouse_down = &WindowScenarioselectMousedown;
-    events.get_scroll_size = &WindowScenarioselectScrollgetsize;
-    events.scroll_mousedown = &WindowScenarioselectScrollmousedown;
-    events.scroll_mouseover = &WindowScenarioselectScrollmouseover;
-    events.invalidate = &WindowScenarioselectInvalidate;
-    events.paint = &WindowScenarioselectPaint;
-    events.scroll_paint = &WindowScenarioselectScrollpaint;
-});
-// clang-format on
+    if (!LocalisationService_UseTrueTypeFont())
+        return 24;
```

Comment:
Hardcoded number - may be nicer to create a named constant (`constexpr`) for this at the top, near where `WW`, `WH`, `SidebarWidth`, `TabWidth`, and `TabHeight` are defined already.